SwrveSDK/SDK/Track/Swrve.m

```diff
@@ -814,10 +814,45 @@ -(id) initWithAppID:(int)swrveAppID apiKey:(NSString*)swrveAPIKey config:(SwrveC
     }
 
     [self sendQueuedEvents];
-
+    
+    // legacy from 4.7
+    [self migrateFileProtection];
+    
     return self;
 }
 
+- (BOOL) isProtectedItemAtPath:(NSString *)path {
+    BOOL            result                      = YES;
+    NSDictionary    *attributes                 = nil;
+    NSString        *protectionAttributeValue   = nil;
+    NSFileManager   *fileManager                = nil;
+    NSError         *error                      = nil;
+    
+    fileManager = [[NSFileManager alloc] init];
+    attributes = [fileManager attributesOfItemAtPath:path error:&error];
+    if (attributes != nil){
+        protectionAttributeValue = [attributes valueForKey:NSFileProtectionKey];
+        if ((protectionAttributeValue == nil) || [protectionAttributeValue isEqualToString:NSFileProtectionNone]){
+            result = NO;
+        }
+    } else {
+        DebugLog(@"There was an issue finding the level of file protection for : %@  \nError: %@" , path , error);
+    }
+    return result;
+}
+
+
+- (void) migrateFileProtection {
+
+    if([self isProtectedItemAtPath:[SwrveFileManagement applicationSupportPath]]){
```

Comment:
The thing about checking a version number is that its the fastest and most efficient way. You don't want to be running these lines of code on every single instantiation of the sdk forever and ever. Most migrations systems work this way.